syntax = "proto3";

package tx_car;

enum DataSource { UserInput = 0; CarSim_PAR = 1; }

// one real axis of 1d or 2d map
message RealAxis {
  string disp_name = 1;
  string comment = 2;
  string unit = 3;
  repeated double data = 15;
  DataSource data_source = 16;
}

// real vector 3
message RealVec3 {
  string disp_name = 1;
  string comment = 2;
  string unit = 3;
  double x = 4;
  double y = 5;
  double z = 6;
  DataSource data_source = 16;
}

// real vector 4
message RealVec4 {
  string disp_name = 1;
  string comment = 2;
  string unit = 3;
  double x = 4;
  double y = 5;
  double z = 6;
  double w = 7;
  DataSource data_source = 16;
}

// real scalar
message RealScalar {
  string disp_name = 1;
  string comment = 2;
  string unit = 3;
  double val = 15;
  DataSource data_source = 16;
}

// real 1d map
message Real1DMap {
  string disp_name = 1;
  string comment = 2;
  RealAxis u0_axis = 3;
  RealAxis y0_axis = 4;
  DataSource data_source = 16;
}

// real 2d map
message Real2DMap {
  string disp_name = 1;
  string comment = 2;
  RealAxis u0_axis = 3;
  RealAxis u1_axis = 4;
  RealAxis y0_axis = 5;
  DataSource data_source = 16;
}

// string variable
message StringVariable {
  string disp_name = 1;
  string comment = 2;
  string str_var = 15;
  DataSource data_source = 16;
}
